libpgcommon/lwgeom_cache.h

```diff
@@ -43,10 +55,8 @@ MemoryContext PostgisCacheContext(FunctionCallInfo fcinfo);
 */
 typedef struct {
 	int                         type;
-	GSERIALIZED*                geom1;
-	GSERIALIZED*                geom2;
-	size_t                      geom1_size;
-	size_t                      geom2_size;
+	SHARED_GSERIALIZED *geom1;
+	SHARED_GSERIALIZED *geom2;
 	LWGEOM*                     lwgeom1;
```

Comment:
Not really your Problem, but I'm pretty sure in one of my rewrites I stripped the `lgeom` entries from this struct as they aren't ever actually used in any of the cache lines.

Reply:
I'll have a look. I did remove the size since that was already included in the gserialized struct itself (just behind the VARSIZE macros), so I'll try to clean the lwgeom pointers too if they are unused.